AbilityChanger/Abilities/SuperDash.cs

```diff
@@ -0,0 +1,169 @@
+        public override string inventoryDescKey { get; protected set; } = "INV_DESC_SUPERDASH";
+        public SuperDash():base()
+        {
+            
+        }
+
+        public override void OnFsmEnable(On.PlayMakerFSM.orig_OnEnable orig, PlayMakerFSM self)
+        {
+            orig(self);
+            if (self.gameObject.name == "Knight" && self.FsmName == "Superdash")
+            {
+                #region Start
+                self.AddFirstAction("Dash Start", new CustomFsmAction(() =>
+                {
+                    HandleStart();
+                }));
+
+                #endregion
+
+                #region Charge
+                self.AddFirstAction("Wall Charge", new CustomFsmActionFixedUpdate(() =>
+                {
+                    HandleCharge();
+                }));
+
+                self.AddFirstAction("Ground Charge", new CustomFsmActionFixedUpdate(() =>
+                {
+                    HandleCharge();
+                }));
+
+                #endregion
+
+                #region Charged
+                self.AddFirstAction("Wall Charged", new CustomFsmActionFixedUpdate(() =>
+                {
+                    HandleCharged();
+                }));
+
+                self.AddFirstAction("Ground Charged", new CustomFsmActionFixedUpdate(() =>
+                {
+                    HandleCharged();
+                }));
+
+                #endregion
+
+                #region Cancel
+                self.AddFirstAction("Air Cancel", new CustomFsmAction(() =>
+                {
+                    HandleCancel();
+                }));
+
+                self.AddFirstAction("Hit Wall", new CustomFsmAction(() =>
+                {
+                    HandleCancel();
+                }));
+                #endregion
+
+                #region Trigger
+                self.Intercept(new TransitionInterceptor()
+                {
+                    fromState="Can Superdash?",
+                    toStateDefault="Relinquish Control",
+                    eventName="FINISHED",
+                    toStateCustom="Inactive",
+                    shouldIntercept=()=> currentAbility.hasTrigger(),
+                    onIntercept=(a,b)=> HandleTrigger()
+
+                });
+
+                #endregion
+
+                #region OnGoing
+                self.AddFirstAction("Dashing",new CustomFsmActionFixedUpdate(() =>
+                {
+                    HandleOngoing();
+                }));
+                self.AddFirstAction("Cancelable", new CustomFsmActionFixedUpdate(() =>
+                {
+                    HandleOngoing();
+                }));
+
+                #endregion
+
+                #region Contact
+
+                #endregion
+
+                #region Complete
+                self.AddFirstAction("Regain Control",new CustomFsmAction(() =>
+                {
+                    HandleComplete();
+                }));
+                #endregion
```

Comment:
adding an action at the start of the state can often break compatibility with other mods, can we move these at the end?